src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,36 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(country);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Couldn't add country " + country, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return country;
     }
 
     @Override
     public Optional<Country> get(Long id) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
```

Comment:
Change to try with resources. (We must close the session)
```suggestion
       try (Session session = factory.openSession()) {
```